postgresql/manifests/postgres.yaml

```diff
@@ -0,0 +1,129 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: postgres
+  annotations:
+    prometheus.io/scrape: "true"
+    prometheus.io/path: /metrics
+    prometheus.io/port: "9187"
+    uw.health.aggregator.enable: "false"
+spec:
+  ports:
+    - name: db
+      port: 5432
+      protocol: TCP
+    - name: exporter
+      protocol: TCP
+      port: 9187
+  selector:
+    app: postgres
+---
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: postgres
+  labels:
+    app: postgres
+spec:
+  serviceName: postgres
+  replicas: 1
+  selector:
+    matchLabels:
+      app: postgres
+  template:
+    metadata:
+      annotations:
+        prometheus.io/scrape: "true"
+        prometheus.io/path:   /__/metrics
+        prometheus.io/port:   "8081"
+      labels:
+        app: postgres
+      name: postgres
+    spec:
+      containers:
+      - name: metrics
+        env:
+        - name: DATA_SOURCE_NAME
+          valueFrom:
+            secretKeyRef:
+              name: postgres-secrets
+              key: url
+        resources:
+          limits:
+            memory: 64Mi
+            cpu: 100m
+        ports:
+          - containerPort: 9187
+            protocol: TCP
+        imagePullPolicy: IfNotPresent
+        image: quay.io/prometheuscommunity/postgres-exporter:0.15.0
+      - name: postgres
+        image: postgres:16-alpine
+        command:
+          - docker-entrypoint.sh
+          - -c
+          - work_mem=512MB
+          - -c
+          - max_parallel_workers_per_gather=0
+          - -c
+          - max_wal_size=2GB
+        imagePullPolicy: IfNotPresent
+        resources:
+          requests:
+            cpu: 0
+            memory: 0
+          limits:
+            memory: 2000Mi
```

Comment:
My k8s-foo is middling, but my understanding of resource handling is that if the request value is lower than the limit, the pod will be started will be allowed to expand its resource ask inside that spread, *but* the provisioner may well deploy a pod to a node that has more RAM than the requested value, but less RAM than the memory limit.  In cases where the memory request grows, the provisioner may kill the pod and re-deploy it to another node where those resources are available.

Normally that's fine with any old service, but with a database, you probably want to prevent that from happening, in which case I'd set the request and limit to the same value.

Also, a 2GB default seems rather high.  Like the disk space ask, it might be best to keep the initial request low and let people scale up as they need.

Reply:
I think it tends to be the case that if you set limits when you don't know where they should be it causes more problems for the provisioner than its worth. 

I will reduce the limit but I think in general you should tune them both when you know where you actually need them